include/albatross/src/models/sparse_gp.hpp

```diff
@@ -317,7 +319,8 @@ class SparseGaussianProcessRegression
       B.col(pi).topRows(i + 1) = old_fit.sigma_R.col(i).topRows(i + 1);
     }
     B.bottomRows(n_new) = A_ldlt.sqrt_solve(K_fu);
-    const auto B_qr = B.colPivHouseholderQr();
+    auto B_qr = SPQR_create(B.sparseView());
```

Comment:
curious how long `SPQR_create` takes, would it be helpful if `B` were already a sparse matrix type?

Reply:
It would definitely be helpful, but it significantly complicates assembling `B`, as you can't just assign to block views of sparse matrices.  I don't mind making that change if you'd like to see it.

I haven't done any benchmarks on this whole integrated system yet; my benchmarks were just on solving sparse problems I'd dumped out from a model already in sparse format.  I'll put some head-to-head test results up for the Swift model that uses a sparse GP in that repo.